File: j2k/tests/testData/ast/constructors/nestedClassNameInParameterDefaults4.java
package pack;

import static pack.A.*;

class A {
    A(Nested nested) {
    }

    A() {
        this(new Nested(Nested.FIELD));
    }

    static class Nested {
        Nested(int p){}

        public static final int FIELD = 0;
    }
}

class B {
    Nested nested;
}

File: j2k/testData/fileOrElement/constructors/nestedClassNameInParameterDefaults4.kt
package pack

import pack.A.*

internal class A @JvmOverloads constructor(nested: Nested = Nested(Nested.FIELD)) {

    internal class Nested(p: Int) {
        companion object {

            val FIELD = 0
        }
    }
}

internal class B {
    var nested: Nested? = null
}